XiEditor/Trace.swift

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A simple mechanism for logging trace events and outputting a file in
+// Chrome tracing format
+
+import Foundation
+
+var globalTrace = Trace()
```

Comment:
Swift style[1] would be to the global be a static type property, ala `NSApplication.shared` or `FileManager.default`. So perhaps `Trace.default`?

A partial solution to your TODO, below, would be for `Trace` to be init'd with a path (maybe just a directory) where it will be saved. 

1. [Adopting Cocoa Design Patterns](https://developer.apple.com/library/content/documentation/Swift/Conceptual/BuildingCocoaApps/AdoptingCocoaDesignPatterns.html) (search 'singleton')

Reply:
I went with `shared`, seems appropriate.